(docs) List which config options are pluggable (#2526)
puppetlabs/bolt: documentation/using_plugins.md

such as `config` or `facts` are resolved once Bolt starts running an action on
that target.

> While plugins are supported for configuration files, they're disable for some settings. Plugins are

Reviewer: ```suggestion
While plugins are supported for configuration files, they're disabled for some settings. Plugins are
```

> enabled for `plugin-hooks`, `plugins`, `puppetdb` and subsettings, and `inventory-config` and any
> subsettings. If you'd like another setting to have plugins enabled, let us know [in #bolt in

Reviewer 2: ```suggestion
enabled for `plugin-hooks`, `plugins`, `puppetdb` and subsettings, and transport configuration
under `inventory-config`. If you'd like another setting to have plugins enabled, let us know [in #bolt in
```

`inventory-config` itself is not pluggable.

> slack](https://slack.puppet.com) or by making an issue.

Reviewer 2: ```suggestion
slack](https://slack.puppet.com) or by [making an issue](https://github.com/puppetlabs/bolt/issues/new/choose).
```

such as `config` or `facts` are resolved once Bolt starts running an action on
that target.

While plugins are supported for configuration files, they're disabled for some settings. You can see
which options are available on configuration references pages for

> [bolt.yaml](bolt_configuration_reference.md), [bolt-project.yaml](bolt_project_reference.md), and

Reviewer: Do we want bolt.yaml here? My vote would be to leave it out cos it's one less thing to whackamole when we hit 3.0

Author: Great point, removed it!
